projects/distributed-tracing/src/shared/dashboard/widgets/table/table-widget-control-checkbox-option.model.ts

```diff
@@ -31,7 +32,10 @@ export class TableWidgetControlCheckboxOptionModel extends TableWidgetControlMod
     );
   }
 
-  private isValidCheckboxControlOption(options: TableControlOption[]): options is TableCheckboxOptions {
-    return options.length === 2 && options.every(option => typeof option.value === 'boolean');
+  private isValidCheckboxControlOptions(options: TableControlOption[]): options is TableCheckboxOptions {
+    return (
+      options.length === 2 &&
+      options.every(option => hasOwnProperty(option, 'value') && typeof option.value === 'boolean')
```

Comment:
I am yet to understand the inheritance in the table control options. value property seems to be defined at only one place. We can also use `in` here, i think

Reply:
Yeah, they are confusing. Fortunately it got a little bit cleaner with this PR.

Correct, that `value` is only for checkbox control now, since it has to hold a `true` or `false` while the select control doesn't. Checkbox controls are actually the oddball because they have to hold more ways of modifying a query. 

Both select and checkbox controls have a `metaValue`. This is what contains the information about what to use in the query. It is most typically a `Filter` (see the `metaValue` for `TableFilterControlOption`), containing `key`, `operator`, and `value`. This is the `metaValue` that all select controls use now.

There are two other cases with query info, but they are now only used for checkbox controls. In addition to the 'Filter' 'metaValue`, there is an `Unset` and a `Property` `metaValue` (see the `metaValue` for `TableUnsetControlOption` and `TablePropertyControlOption`). These are also used to modify the query, but in a different way.

The `Unset` `metaValue` is used to hold a `string` that identifies which filter to _remove_ from our filter list. The `Property` `metaValue` is a `Dictionary` used to add request properties to the query. Currently, our only example of this is `includeInactive: true/false` that we add to the request for APIs.

So, the reason that checkbox has a `value` and not the select dropdown is that checkbox has this additional user specified `true/false` in addition to the `value` that is buried in the `metaValue` for all controls.

Not sure if that explains it or makes it more confusing, but happy to expand on anything.